### PR TITLE
Create a test Cardano transactions builder

### LINKED ISSUE #1667: Create a test Cardano transactions builder
## Why
We need to have a simple and versatile way of creating test transactions which are used in our unit tests. We have already created multiple generators in the code that will be superseded by this builder. The builder will be tested which will allow better quality of tests.

## What
Assess the existing generators and implement a builder that will fit all the existing needs and which can be easily extended with new needs.

## How

- [x] Assess the existing generators (Focus on the generator in the [prover service](https://github.com/input-output-hk/mithril/blob/481b7e782fe83b7dcfed8ef23baedb3b0907e512/mithril-aggregator/src/services/prover.rs#L188))
- [x] Design and implement a transactions builder

### PR DESCRIPTION
**Pull request by dlachaume**
## Content
This PR adds the `CardanoTransactionsBuilder`, which aims to easily build transactions for test cases.

## Pre-submit checklist

- Branch
  - [x] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [ ] CHANGELOG file is updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested
- Documentation
  - [ ] Update README file (if relevant)
  - [ ] Update documentation website (if relevant)
  - [ ] Add dev blog post (if relevant)

## Issue(s)
Closes #1667

### CODE REVIEW

**Review by jpraynaud — COMMENTED** [review]
Overall looks good :+1: 

I left some comments and suggestions below.

**Review by Alenar — APPROVED** [review]
LGTM

File: mithril-common/src/test_utils/cardano_transactions_builder.rs [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+            block_number,
+            transaction_id,
+            format!("block-hash-{block_number}"),
+            immutable_file_number,
+        )
+    }
+}
+
+#[cfg(test)]
+mod test {
```
[the remark this answers is not in the document]
Comment (dlachaume) [reply mithril-common/src/test_utils/cardano_transactions_builder.rs:178]: We've added examples tests in the rust doc.

File: internal/mithril-persistence/src/database/repository/cardano_transaction_repository.rs [hunk 2]
```diff
@@ -871,21 +895,18 @@ mod tests {
...
-                block_number: i,
-                slot_number: i * 100,
-                block_hash: format!("block-hash-{i}"),
-                immutable_file_number: 1,
-            })
+        let cardano_transactions: Vec<CardanoTransactionRecord> = CardanoTransactionsBuilder::new()
+            .blocks_per_block_range(15)
+            .build_transactions(53)
+            .iter()
+            .map(|transaction| CardanoTransactionRecord::from(transaction.clone()))
```
[the remark this answers is not in the document]
Comment (dlachaume) [reply internal/mithril-persistence/src/database/repository/cardano_transaction_repository.rs]: Done ✅

**Review by jpraynaud — APPROVED** [review]
LGTM :+1:

**Pull request merged**